src/Waves/Signer.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+
+#include "../Hash.h"
+
+using namespace TW;
+using namespace TW::Waves;
+
+Data Signer::sign(const PrivateKey &privateKey, Transaction &transaction) noexcept {
+    auto bytesToSign = transaction.serializeToSign();
```

Comment:
`serializeToSign` might throw exception, can we do `try catch` and return empty data?